src/awe/binarchive.cpp

```diff
@@ -60,10 +61,21 @@ Common::ReadStream *BINArchive::getResource(const std::string &rid) const {
 }
 
 void BINArchive::load(Common::ReadStream &bin) {
+	bin.seek(0);
+	std::string magic = bin.readFixedSizeString(4);
```

Comment:
Normally we use for this 4 character codes the MKTAG macro and the reading of a big endian integer. You can look at for example the FSBFile class as it is done there.